src/main/java/com/jcraft/jsch/Compression.java

```diff
@@ -32,7 +32,7 @@
 public interface Compression{
   static public final int INFLATER=0;
   static public final int DEFLATER=1;
-  void init(int type, int level) throws Exception;
+  void init(Session session, int type, int level) throws Exception;
```

Comment:
In #126 I've attempted to restore the backwards incompatibilities I introduced in several public interfaces.

Could we change revert this method signature back, and then instead add something like this?
```
default void init(int type, int level, Session session) throws Exception {init(type, level);}
default Logger getLogger() {return JSch.getLogger();}
```

That way this interface maintains backwards compatibility.
We can thin change the built-in implementations to override these methods as appropriate, but any users that are using custom `Compression` implementations don't suddenly have their implementations break?

Reply:
I've reverted the change but solved it a bit differently to keep code duplication in juz.Compression and jzlib.Compression to a minimum. There is now a default setSession-method doing nothing and a static logMessage in Session with a Supplier<String> that is now called in both Compression-implementations.